benchmarks/low_level.cpp

```diff
@@ -0,0 +1,455 @@
+// Copyright (c) 2021 MIT Digital Currency Initiative,
+//                    Federal Reserve Bank of Boston
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+/*
+    File contains benchmarks for
+    -   transactions size [1, 4] -> [1, 2]
+    -   full tx serialization
+    -   full tx deserialization
+    -   full tx round trip
+    -   compact tx serialization
+    -   compact tx deserialization
+    -   compact tx round trip
+    -   tx creation
+    -   tx signing
+    -   tx validation
+
+
+    Note: Contains call to BENCHMARK_MAIN
+*/
+
+#include "uhs/transaction/messages.hpp"
+#include "uhs/transaction/transaction.hpp"
+#include "uhs/transaction/validation.hpp"
+#include "uhs/transaction/wallet.hpp"
+
+#include "util/serialization/istream_serializer.hpp"
+#include "util/serialization/ostream_serializer.hpp"
+
+#include <benchmark/benchmark.h>
+#include <filesystem>
+#include <gtest/gtest.h>
+
+class low_level : public ::benchmark::Fixture {
+  protected:
+    void SetUp(const ::benchmark::State&) override {
+        auto mint_tx1 = wallet1.mint_new_coins(2, 100);
+        wallet1.confirm_transaction(mint_tx1);
+        auto mint_tx2 = wallet2.mint_new_coins(1, 100);
+        wallet2.confirm_transaction(mint_tx2);
+
+        m_of.open(m_BENCHMARK_File);
+    }
+
+    void TearDown(const ::benchmark::State&) override {
+        m_if.close();
+        m_of.close();
+        std::filesystem::remove_all(m_BENCHMARK_File);
+    }
+
+    cbdc::transaction::wallet wallet1;
+    cbdc::transaction::wallet wallet2;
+
+    cbdc::transaction::full_tx m_valid_tx{};
+    cbdc::transaction::full_tx m_valid_tx_multi_inp{};
+
+    // std::unique_ptr<secp256k1_context, decltype(&secp256k1_context_destroy)>
```

Comment:
Here, and elsewhere, let's avoid leaving dangling comments.

Reply:
Good note -- mistake on my part, I thought I had removed these interleaved comments before sending this commit